## import necessary library

In [1]:
import pandas as pd

### Task 1 merge all 12 months sales data into 1 csv file

In [3]:
import glob

path = r'./Sales_Data'
all_files = glob.glob(path + "/*.csv")
all_files

['./Sales_Data/Sales_December_2019.csv',
 './Sales_Data/Sales_April_2019.csv',
 './Sales_Data/Sales_February_2019.csv',
 './Sales_Data/Sales_March_2019.csv',
 './Sales_Data/Sales_August_2019.csv',
 './Sales_Data/Sales_May_2019.csv',
 './Sales_Data/Sales_November_2019.csv',
 './Sales_Data/Sales_October_2019.csv',
 './Sales_Data/Sales_January_2019.csv',
 './Sales_Data/Sales_September_2019.csv',
 './Sales_Data/Sales_July_2019.csv',
 './Sales_Data/Sales_June_2019.csv']

In [14]:
for f in all_files:
    df = pd.read_csv(f)
    print(df.shape)

(25117, 6)
(18383, 6)
(12036, 6)
(15226, 6)
(12011, 6)
(16635, 6)
(17661, 6)
(20379, 6)
(9723, 6)
(11686, 6)
(14371, 6)
(13622, 6)


In [27]:
df_from_each_file = (pd.read_csv(f, sep=',') for f in all_files)
df_merged = pd.concat(df_from_each_file, ignore_index=True)
df_merged.to_csv("merged.csv", index=False)

In [30]:
# take a look at the merged csv
merged = pd.read_csv('merged.csv')
#merged.reset_index(drop=True, inplace=True)
merged.shape

(186850, 6)

In [31]:
merged.head(10)

Order ID                     Product Quantity Ordered Price Each  \
0   295665          Macbook Pro Laptop                1       1700   
1   295666          LG Washing Machine                1      600.0   
2   295667        USB-C Charging Cable                1      11.95   
3   295668            27in FHD Monitor                1     149.99   
4   295669        USB-C Charging Cable                1      11.95   
5   295670       AA Batteries (4-pack)                1       3.84   
6   295671        USB-C Charging Cable                1      11.95   
7   295672        USB-C Charging Cable                2      11.95   
8   295673  Bose SoundSport Headphones                1      99.99   
9   295674      AAA Batteries (4-pack)                4       2.99   

       Order Date                           Purchase Address  
0  12/30/19 00:01     136 Church St, New York City, NY 10001  
1  12/29/19 07:03        562 2nd St, New York City, NY 10001  
2  12/12/19 18:21       277 Main St, New York City, NY 10001  
3  12/22/19 15:13        410 6th St, San Francisco, CA 94016  
4  12/18/19 12:38              43 Hill St, Atlanta, GA 30301  
5  12/31/19 22:58  200 Jefferson St, New York City, NY 10001  
6  12/16/19 15:10            928 12th St, Portland, OR 97035  
7  12/13/19 09:29           813 Hickory St, Dallas, TX 75001  
8  12/15/19 23:26            718 Wilson St, Dallas, TX 75001  
9  12/28/19 11:51                77 7th St, Dallas, TX 75001

### Drop NAN rows

In [48]:
merged.dropna(how='all', inplace=True)
#

### Question1 What was the best month for sales? How much was earned that month?

### Add Month Column

In [55]:
merged['Month of Order'] = merged['Order Date'].str[:2]
merged = merged[merged['Month of Order'] != 'Or']
merged.reset_index()
merged['Month of Order'] = merged['Month of Order'].astype('int32')
merged.head(300)

Order ID                 Product Quantity Ordered Price Each  \
0     295665      Macbook Pro Laptop                1       1700   
1     295666      LG Washing Machine                1      600.0   
2     295667    USB-C Charging Cable                1      11.95   
3     295668        27in FHD Monitor                1     149.99   
4     295669    USB-C Charging Cable                1      11.95   
..       ...                     ...              ...        ...   
297   295941         ThinkPad Laptop                1     999.99   
298   295942            Google Phone                1        600   
299   295943  AAA Batteries (4-pack)                3       2.99   
300   295944  27in 4K Gaming Monitor                1     389.99   
301   295945        Wired Headphones                1      11.99   

         Order Date                        Purchase Address  Month of Order  
0    12/30/19 00:01  136 Church St, New York City, NY 10001              12  
1    12/29/19 07:03     562 2nd St, New York City, NY 10001              12  
2    12/12/19 18:21    277 Main St, New York City, NY 10001              12  
3    12/22/19 15:13     410 6th St, San Francisco, CA 94016              12  
4    12/18/19 12:38           43 Hill St, Atlanta, GA 30301              12  
..              ...                                     ...             ...  
297  12/31/19 16:24       64 Dogwood St, Portland, OR 97035              12  
298  12/30/19 19:21          150 North St, Boston, MA 02215              12  
299  12/04/19 13:32         701 Sunset St, Dallas, TX 75001              12  
300  12/29/19 17:04    175 Main St, San Francisco, CA 94016              12  
301  12/06/19 10:20         207 Sunset St, Austin, TX 73301              12  

[300 rows x 7 columns]

In [57]:
merged['Price Each'] = pd.to_numeric(merged['Price Each'])
merged['Quantity Ordered'] = pd.to_numeric(merged['Quantity Ordered'])


In [58]:
merged['Total Price'] = merged['Price Each']*merged['Quantity Ordered']
merged.head()

Order ID               Product  Quantity Ordered  Price Each  \
0   295665    Macbook Pro Laptop                 1     1700.00   
1   295666    LG Washing Machine                 1      600.00   
2   295667  USB-C Charging Cable                 1       11.95   
3   295668      27in FHD Monitor                 1      149.99   
4   295669  USB-C Charging Cable                 1       11.95   

       Order Date                        Purchase Address  Month of Order  \
0  12/30/19 00:01  136 Church St, New York City, NY 10001              12   
1  12/29/19 07:03     562 2nd St, New York City, NY 10001              12   
2  12/12/19 18:21    277 Main St, New York City, NY 10001              12   
3  12/22/19 15:13     410 6th St, San Francisco, CA 94016              12   
4  12/18/19 12:38           43 Hill St, Atlanta, GA 30301              12   

   Total Price  
0      1700.00  
1       600.00  
2        11.95  
3       149.99  
4        11.95

In [68]:
sales_per_month = merged.groupby('Month of Order')['Total Price'].sum()
sales_per_month.sort_values(ascending=False)

Month of Order
12    4613443.34
10    3736726.88
4     3390670.24
11    3199603.20
5     3152606.75
3     2807100.38
7     2647775.76
6     2577802.26
8     2244467.88
2     2202022.42
9     2097560.13
1     1822256.73
Name: Total Price, dtype: float64

### So December is the month with highest sales with $4613443.34 sales